File: mc-plugins/lores/src/main/java/org/lores/Lores.java
package org.lores;

import org.bukkit.plugin.java.JavaPlugin;
import org.lores.commands.Lore;

import java.util.Objects;
import java.util.Properties;

/**
 * @author yanni
 */
public class Lores extends JavaPlugin {


    @Override
    public void onEnable() {
        registerCommands();
        Properties version = new Properties();
        try {
            version.load(getResource("version.properties"));
        } catch (Exception e) {
            getLogger().info("error find version");
        }
        getLogger().info("Lores " + getDescription().getVersion() + " (Build " + version.getProperty("Build") + ") is enabled!");
    }

    public void registerCommands() {
        Objects.requireNonNull(this.getCommand("lore")).setExecutor(new Lore(this));
        Objects.requireNonNull(this.getCommand("lore")).setTabCompleter(new Lore(this));
    }
}

File: mcplugins-kt/lores/src/main/kotlin/org/lores/Lores.kt
package org.lores

import org.bukkit.plugin.java.JavaPlugin
import java.util.*

/**
 * @author yanni
 */
class Lores : JavaPlugin() {
    override fun onEnable() {
        registerCommands()
        val version = Properties()
        try {
            version.load(getResource("version.properties"))
        } catch (e: Exception) {
            logger.info("error find version")
        }
        logger.info("Lores " + description.version + " (Build " + version.getProperty("Build") + ") is enabled!")
    }

    private fun registerCommands() {
        getCommand("lore")?.setExecutor(LoreCommand(this))
        getCommand("lore")?.tabCompleter = LoreCommand(this)
    }
}